redbot/core/bot.py

```diff
@@ -1240,10 +1247,15 @@ async def get_command_setting(guild_id: int) -> Optional[bool]:
             scope = self._config.custom(COMMAND_SCOPE, command.qualified_name, guild_id)
             return await scope.embeds()
 
+        if isinstance(channel, discord.GroupChannel) or isinstance(channel, discord.DMChannel):
+            # we should not be passing group channels to this function
+            # this only uses global settings
+            raise TypeError("You cannot pass a GroupChannel or DMChannel to this method")
+
         # using dpy_commands.Context to keep the Messageable contract in full
         if isinstance(channel, dpy_commands.Context):
             command = command or channel.command
-            channel = channel.channel
+            channel = channel.author
```

Comment:
That's not right for Context with any other `Context.channel` than `DMChannel` such as `TextChannel`